install_subrepos.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+
+"""
+Helper script for installing spyder and external-deps locally in editable mode.
+"""
+
+import argparse
+import os
+import re
+import sys
+from logging import Formatter, StreamHandler, getLogger
+from pathlib import Path
+from pkg_resources import DistributionNotFound, get_distribution
+from subprocess import check_output
+
+DEVPATH = Path(__file__).resolve().parent
+DEPS_PATH = DEVPATH / 'external-deps'
+BASE_COMMAND = [sys.executable, '-m', 'pip', 'install', '--no-deps']
+
+REPOS = {}
+for p in DEPS_PATH.iterdir():
+    if p.name.startswith('.'):
+        continue
+    try:
+        dist = Path(get_distribution(p.name).location)
+    except DistributionNotFound:
+        dist = None
+
+    REPOS[p.name] = {'repo': p, 'dist': dist, 'editable': p == dist}
```

Comment:
```suggestion
    REPOS[p.name] = {
        'repo': p, 'dist': dist, 'editable': p == dist or p in dist.parents}
```


The previous logic is rather fragile and only works in a particular case, when the top-level import package(s) are located directly in the project directory (the "implicit" layout), rather than the recommended "src" layout or any other case. This tweaked logic fixes that.